src/geo/transform.js

```diff
@@ -1417,15 +1417,56 @@ class Transform {
         return new LngLatBounds(new LngLat(west, south), new LngLat(east, north));
     }
 
+    // Gets minimum bounding lngLats for non-Mercator and non-globe projections.
+    _getBoundsProjection(): LngLatBounds {
+        assert(!this.projection.supportsTerrain, "This function doesn't account for terrain");
+        assert(!this.projection.supportsWorldCopies, "Projections that support world copies are rectangular and should the simpler _getProjection");
+        const {top, left} = this._edgeInsets;
+        const bottom = this.height - this._edgeInsets.bottom;
+        const right = this.width - this._edgeInsets.right;
+
+        const res = 20; // Number of points to sample along each edge
+        const w = right - left;
+        const h = bottom - top;
+
+        let west = 180;
+        let east = -180;
+        let south = 90;
+        let north = -90;
+
+        const checkExtreme = (x: number, y: number) => {
+            const {lng, lat} = this.pointLocation(new Point(x, y));
+            west = Math.min(west, lng);
+            east = Math.max(east, lng);
+            south = Math.min(south, lat);
+            north = Math.max(north, lat);
+        };
+
+        for (let i = 0; i <= res; i++) {
+            checkExtreme(left + w * i / res, top);
+            checkExtreme(right, top + h * i / res);
+            checkExtreme(right - w * i / res, bottom);
+            checkExtreme(left, bottom - h * i / res);
+        }
+
+        return new LngLatBounds(new LngLat(west, south), new LngLat(east, north));
+    }
+
     _getBounds(min: number, max: number): LngLatBounds {
         if (this.projection.name === 'globe') {
             return this._getGlobeBounds();
         }
 
-        const topLeft = new Point(this._edgeInsets.left, this._edgeInsets.top);
-        const topRight = new Point(this.width - this._edgeInsets.right, this._edgeInsets.top);
-        const bottomRight = new Point(this.width - this._edgeInsets.right, this.height - this._edgeInsets.bottom);
-        const bottomLeft = new Point(this._edgeInsets.left, this.height - this._edgeInsets.bottom);
+        assert(this.projection.supportsWorldCopies, "_getBounds only accounst for corners and thus needs a rectangular projection. Other projections should use _getBoundsProjection or _getGlobeBounds");
```

Comment:
nit:
```suggestion
        assert(this.projection.supportsWorldCopies, "_getBounds only accounts for corners and thus needs a rectangular projection. Other projections should use _getBoundsProjection or _getGlobeBounds");
```

Also, it seems like `equirectangular` + `mercator` are the only cases here? Maybe ok to inline them to not have to think about which projection do support world copies and fall through this code path.

Reply:
My thinking was that currently supporting world copies is the same as being rectangular, and this is likely to remain the case if we add new projections and provide a slightly more future-proof assertion. (On the other hand, it could be conceivable for conic projections could support world copies.)

But maybe I'm overthinking it!